src/wp-includes/media.php

```diff
@@ -1813,6 +1826,19 @@ function wp_filter_content_tags( $content, $context = null ) {
 		}
 	}
 
+	foreach ( $iframes as $iframe => $attachment_id ) {
+		$filtered_iframe = $iframe;
+
+		// Add 'loading' attribute if applicable.
+		if ( $add_iframe_loading_attr && false === strpos( $filtered_iframe, ' loading=' ) ) {
+			$filtered_iframe = wp_iframe_tag_add_loading_attr( $filtered_iframe, $context );
+		}
+
+		if ( $filtered_iframe !== $iframe ) {
+			$content = str_replace( $iframe, $filtered_iframe, $content );
```

Comment:
I noticed that `preg_match_all()` is being used, and then the matches have to get found again to be replaced by `str_replace()`. Would it not be better to use `preg_replace_callback()`? Then you'd only need to loop over the matches once and replace them as you go.

Reply:
That's needed for images, to "warm" the attachments cache, see `_prime_post_caches( $attachment_ids, false, true );` above (in the expanded code).

It seems quite faster to do one preg_match/replace instead of two in most cases as `str_replace()` is quite fast and there generally aren't many iframes. Perhaps worth another "speed test" with different content length, number of images and iframes :)

Reply:
I tried using `preg_replace_callback` instead of the `foreach` loop with `str_replace`, the code I used was the following:
```php
	if ( $add_iframe_loading_attr ) {
		$content = preg_replace_callback(
			'/<iframe\s[^>]+>/',
			function( $matches ) use ( $context ) {
				$iframe = $matches[0];
				// Add 'loading' attribute if applicable.
				if ( false === strpos( $iframe, ' loading=' ) ) {
					$iframe = wp_iframe_tag_add_loading_attr( $iframe, $context );
				}
				return $iframe;
			},
			$content
		);
	}
```

I ran a test for 50000 `iframe` tags in the content, and the currently present `foreach` with `str_replace` approach showed to be a lot faster:
* `foreach` with `str_replace` took on average **27.6** ms
* `preg_replace_callback` took on average **195.9** ms

So we should be good with the existing approach. Nice side-effect is that it aligns more nicely with the approach for `img` tags, making the code easier to follow than if we were using a different technique for `iframe` tags.